keyboards/hardwareabstraction/handwire/keymaps/default/keymap.c

```diff
@@ -0,0 +1,334 @@
+    };
+
+    void animation_phase(void) {
+        if (get_current_wpm() <= IDLE_SPEED) {          // Idle edge case
+            current_idle_frame = (current_idle_frame + 1) % IDLE_FRAMES;
+            oled_write_raw_P(idle[abs((IDLE_FRAMES - 1) - current_idle_frame)], ANIM_SIZE);
+        }
+        else if (get_current_wpm() > IDLE_SPEED && get_current_wpm() < TAP_SPEED) {     // Prep frame
+            // oled_write_raw_P(prep[abs((PREP_FRAMES-1)-current_prep_frame)], ANIM_SIZE); // uncomment if IDLE_FRAMES >1
+            oled_write_raw_P(prep[0], ANIM_SIZE);  // remove if IDLE_FRAMES >1
+        }
+        else if (get_current_wpm() >= TAP_SPEED) {
+            current_tap_frame = (current_tap_frame + 1) % TAP_FRAMES;
+            oled_write_raw_P(tap[abs((TAP_FRAMES - 1) - current_tap_frame)], ANIM_SIZE);
+        }
+    }
+
+    if (get_current_wpm() != 000){       // Check for typing
+        oled_on();
+        if (timer_elapsed32(anim_timer) > ANIM_FRAME_DURATION) {
+            anim_timer = timer_read32();
+            animation_phase();
+        }
+        anim_sleep = timer_read32();
+    }
+    else
+    {
+        if (timer_elapsed32(anim_sleep) > OLED_TIMEOUT){         // Check timer for OLED timeout
+            oled_off();
+        } else {
+            if (timer_elapsed32(anim_timer) > ANIM_FRAME_DURATION) {
+                anim_timer = timer_read32();
+                animation_phase();
+            }
+        }
+    }
+}
+
+// Used to display current layer number on OLED
+layer_state_t layer_state_set_user(layer_state_t state){
+    layer = get_highest_layer(state);
+    return state;
+}
+
+// Initialize code to rotate the OLED 180 degrees
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    return OLED_ROTATION_180;  // Flip for headers on right side
+}
+
+// OLED Task
+bool oled_task_user(void) {
+    if(initial_keypress){
+        render_anim();                                    // Render the animation phases
+
+        // Display WPM
+        oled_set_cursor(0, 0);                            // Sets cursor to (column, row) using charactar spacing (5 rows on 128x32 screen, anything more will overflow back to the top)
+        sprintf(wpm_str, "WPM:%03d", get_current_wpm());  // Edit the string to change wwhat shows up, edit %03d to change how many digits show up
```

Comment:
You should use the `oled_write_P` function when possible, as it uses flash rather than system memory, which can be important on AVR based boards. 

Also, sprintf/snprintf is rather expensive (~1.5k). We added a helper function mostly for oleds to help reduce the size: 
https://docs.qmk.fm/#/squeezing_avr?id=oled-tweaks

```suggestion
        oled_write_P(PSTR("WPM:"), false);
        oled_write(get_u8_str(get_current_wpm(), '0'), false);
```